### PR TITLE
Ably 1.2 examples 

### PR DESCRIPTION
**Pull request by sacOO7**
- Added rest and real-time examples for SDK
- Updated README for how to run

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# How to run
+
+1. Realtime
+
+- Go to realtime dir `cd realtime`
+- Rename **.env.example** to **.env**, update `ABLY_KEY` with your own key 
+- Realtime channel pub-sub
+
+    ```go run presence.go constants.go```
```
Comment (tcard) [inline examples/README.md:9]: Inline code snippets are enclosed in single quotes,

    `like this`
    
Triple backticks are for code blocks, and go in their own line.

    ```
    like this 
    ```

([Source](https://guides.github.com/features/mastering-markdown))

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+- Realtime channel presence
+
+    ```go run pub-sub.go constants.go```
+    
+2. Rest
```
Comment (tcard) [inline examples/README.md:16]: This is an acronym.
```suggestion
2. REST
```

File: examples/go.mod [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module examples
+
+go 1.14
+
+require (
+	github.com/ably/ably-go v1.2.0-apipreview.3
```
Comment (tcard) [inline examples/go.mod:6]: Please run `go mod tidy` to apply automatic formatting and other automatic fixes to `go.mod` and `go.sum`.

Also, we're at `v1.2.0-apipreview.4` so this probably should be based on that one.

**Review by tcard — APPROVED** [review]

File: examples/go.mod (hunk 3 above)
Comment (sacOO7) [reply examples/go.mod:6]: Thanks, will make the change

**Comment by sacOO7** [comment]
Merging the code, since it only contains examples : )

**Pull request merged**